### PR TITLE
Popup: move ResizeObserver subscription to renderContent (T1130045)

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: js/ui/popup/ui.popup.js [hunk 1]
```diff
@@ -347,6 +347,11 @@ const Popup = Overlay.inherit({
                 && parseInt(contentRect.height, 10) === this._renderedDimensions?.height;
     },
 
+    _renderContent() {
+        this.callBase();
+        this._observeContentResize(this.option('visible'));
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js]: When do we unsubscribe RO? Seems like `_renderContent` is called only on showing.

File: js/ui/overlay/ui.overlay.js [hunk 2]
```diff
@@ -859,6 +858,8 @@ const Overlay = Widget.inherit({
             transclude,
             onRendered: () => {
                 whenContentRendered.resolve();
+
+                // NOTE: T1114344, T1130045, T390360, T386038
```
Comment (ksercs) [inline js/ui/overlay/ui.overlay.js]: Are really all these tickets related to this row of code?

**Comment by ksercs** [comment]
@marker-dao what about test?

File: js/ui/popup/ui.popup.js (hunk 1 above)
Comment (marker-dao) [reply js/ui/popup/ui.popup.js]: I guess we can unsubscribe [here](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/js/ui/popup/ui.popup.js#L429).

File: js/ui/overlay/ui.overlay.js (hunk 2 above)
Comment (marker-dao) [reply js/ui/overlay/ui.overlay.js]: Move notes: [1](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/js/ui/popup/ui.popup.js#L428), [2](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/js/ui/popup/ui.popup.js#L352), [3](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/js/ui/overlay/ui.overlay.js#L502), [4](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/js/ui/overlay/ui.overlay.js#L863).

**Comment by marker-dao** [comment]
> 

Add TestCafe test [here](https://github.com/DevExpress/DevExtreme/blob/180d1d8287ef5eda0f716edd1d065fd326239a89/testing/testcafe/tests/editors/dropDownBox/popup.ts).

File: testing/testcafe/tests/editors/dropDownBox/popup.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { createScreenshotsComparer } from 'devextreme-screenshot-comparer';
+import url from '../../../helpers/getPageUrl';
+import createWidget from '../../../helpers/createWidget';
+import { safeSizeTest } from '../../../helpers/safeSizeTest';
+
+const BUTTON_CLASS = 'dx-dropdowneditor-button';
+
+fixture`Drop Down Button's Popup`
```
Comment (ksercs) [inline testing/testcafe/tests/editors/dropDownBox/popup.ts]:
```suggestion
fixture`Drop Down Box's Popup`
```

File: js/ui/popup/ui.popup.js [hunk 4]
```diff
@@ -347,6 +347,12 @@ const Popup = Overlay.inherit({
                 && parseInt(contentRect.height, 10) === this._renderedDimensions?.height;
     },
 
+    _renderContent() {
+        this.callBase();
+        // NOTE: T1130045
+        this._observeContentResize(this.option('visible'));
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js]:
```suggestion
        this._observeContentResize(true);
```

File: js/ui/popup/ui.popup.js [hunk 5]
```diff
@@ -419,7 +425,10 @@ const Popup = Overlay.inherit({
     },
 
     _renderVisibilityAnimate: function(visible) {
-        this._observeContentResize(visible);
+        // NOTE: T1130045
+        if(!visible) {
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js]: Couldn't we unsubscribe in another place to get rid of "if" condition? For example, `_hide` method?

File: testing/testcafe/tests/editors/dropDownBox/popup.ts (hunk 3 above)
Comment (marker-dao) [reply testing/testcafe/tests/editors/dropDownBox/popup.ts]: [Fix](https://github.com/DevExpress/DevExtreme/blob/85588cf6448360a32ce9f12134dd6d362059ef84/testing/testcafe/tests/editors/dropDownBox/popup.ts#L8)

File: js/ui/popup/ui.popup.js (hunk 4 above)
Comment (marker-dao) [reply js/ui/popup/ui.popup.js]: [Fix](https://github.com/DevExpress/DevExtreme/blob/85588cf6448360a32ce9f12134dd6d362059ef84/js/ui/popup/ui.popup.js#L353)

File: js/ui/popup/ui.popup.js (hunk 5 above)
Comment (marker-dao) [reply js/ui/popup/ui.popup.js]: Move unsubscription to `_hide` method [here](https://github.com/DevExpress/DevExtreme/blob/85588cf6448360a32ce9f12134dd6d362059ef84/js/ui/popup/ui.popup.js#L431).

File: js/ui/popup/ui.popup.js [hunk 6]
```diff
@@ -347,6 +347,12 @@ const Popup = Overlay.inherit({
                 && parseInt(contentRect.height, 10) === this._renderedDimensions?.height;
     },
 
+    _renderContent() {
+        this.callBase();
+        // NOTE: T1130045
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js]:
```suggestion
        // NOTE: This observe should not be called before async showing is called. See T1130045.
```

File: js/ui/popup/ui.popup.js [hunk 7]
```diff
@@ -419,11 +425,16 @@ const Popup = Overlay.inherit({
     },
 
     _renderVisibilityAnimate: function(visible) {
-        this._observeContentResize(visible);
-
         return this.callBase(visible);
     },
 
+    _hide() {
+        // NOTE: T1130045
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js]: This row is not related to bug fix. If we change it, bug will still be fixed.
```suggestion
```

**Review by Zedwag — APPROVED** [review]

**Review by ksercs — APPROVED** [review]
